crates/wasmtime/src/externals.rs

```diff
@@ -864,6 +894,51 @@ impl Memory {
         MemoryType::from_wasmtime_memory(&self.wasmtime_export.memory.memory)
     }
 
+    /// Safely reads memory contents at the given offset into a buffer.
+    ///
+    /// The entire buffer will be filled.
+    ///
+    /// If the offset + buffer length would exceed the current memory capacity,
+    /// a [`MemoryAccessError`] is returned.
+    pub fn read(&self, offset: usize, buffer: &mut [u8]) -> Result<(), MemoryAccessError> {
+        // Guard against overflows.
```

Comment:
It's ok to remove this comment, I don't think it adds much over what the existence of the `checked_add` already implies.

Reply:
Although thinking about this again, another perhaps safer way to write these methods would be:

```rust
let src = self.data_unchecked().get(offset..offset + buffer.len())?;
buffer.copy_from_slice(src);
Ok(())
```

(and something similar below)

Reply:
Or actually to avoid having to do checked arithmetic:

```rust
let src = self.data_unchecked().get(offset..).get(..buffer.len())?;
```